airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py

```diff
@@ -0,0 +1,205 @@
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        zip_file = zipfile.ZipFile(io.BytesIO(response.content))
+        for gzip_filename in zip_file.namelist():
+            with zip_file.open(gzip_filename) as file:
+                yield from self._parse_zip_file(file)
+
+    def _parse_zip_file(self, zip_file: IO[bytes]) -> Iterable[Mapping]:
+        with gzip.open(zip_file) as file:
+            for record in file:
+                yield json.loads(record)
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
+        params = self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=None)
+
+        # API returns data only when requested with a difference between 'start' and 'end' of 6 or more hours.
+        if pendulum.parse(params["start"]).add(hours=6) > pendulum.parse(params["end"]):
+            return []
+        yield from super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+        if stream_state or next_page_token:
+            params["start"] = pendulum.parse(params["start"]).add(hours=1).strftime(self.date_template)
+        return params
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"/{self.api_version}/export"
+
+
+class ActiveUsers(IncrementalAmplitudeStream):
+    base_params = {"m": "active", "i": 1, "g": "country"}
+    name = "active_users"
+    primary_key = "date"
+    time_interval = {"months": 1}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_data = response.json().get("data", [])
+        if response_data:
+            series = list(map(list, zip(*response_data["series"])))
+            for i, date in enumerate(response_data["xValues"]):
+                yield {"date": date, "statistics": dict(zip(response_data["seriesLabels"], series[i]))}
+
+    def path(self, **kwargs) -> str:
+        return f"/{self.api_version}/users"
+
+
+class AverageSessionLength(IncrementalAmplitudeStream):
+    name = "average_session_length"
+    primary_key = "date"
+    time_interval = {"days": 15}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_data = response.json().get("data", [])
+        if response_data:
+            series = response_data["series"][0]
```

Comment:
why are we getting the first index in the series list? Can you add a comment explaining it? It's not obvious from the API docs there will be only one record I think https://developers.amplitude.com/docs/dashboard-rest-api#average-session-length

Reply:
Added comment